type-generator/src/test/scala/mesosphere/raml/RamlTypeGeneratorTest.scala

```diff
@@ -0,0 +1,25 @@
+package mesosphere.raml
+
+import java.io.File
+
+import org.raml.v2.api.RamlModelBuilder
+import org.scalatest.Matchers._
+import org.scalatest.{GivenWhenThen, WordSpec}
+
+class RamlTypeGeneratorTest extends WordSpec with GivenWhenThen {
+
+  "RamlTypeGenerator" should {
+    "produce types" in {
+      Given("A reference model")
+      val file = new File("src/test/resrouces/Test.raml")
+      val models = Seq(new RamlModelBuilder().buildApi(file))
+
+      When("the type is generated")
+      val types = RamlTypeGenerator(models.toVector, "mesosphere.raml.test")
+      val typesAsStr: Int = types.mapValues(treehugger.forest.treeToString(_))
```

Comment:
This is of course the wrong type. Intellij is not recognizing the project yet so I have to investigate the types a little more.